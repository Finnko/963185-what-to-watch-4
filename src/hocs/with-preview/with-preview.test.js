import React from "react";
import renderer from "react-test-renderer";
import withPreview from './with-preview';
import withVideo from '../with-video/with-video';
import VideoPlayer from '../../components/video-player/video-player.jsx';

const props = {
  isPlaying: false,
  videoSrc: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`,
  preview: `img/the-grand-budapest-hotel-poster.jpg`,
};

const VideoPlayerWrapped = withVideo(VideoPlayer);

const MockComponentWrapped = withPreview(VideoPlayerWrapped);

describe(`withPreview works correctly`, () => {
  it(`withPreview is rendered correctly`, () => {

    const tree = renderer.create((
      <MockComponentWrapped
        {...props}
        onMovieMouseEnter={() => {}}
        onMovieMouseLeave={() => {}}
        renderPlayer={() => {}}
      />
    ), {
      createNodeMock() {
        return {};
      }
    }).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
