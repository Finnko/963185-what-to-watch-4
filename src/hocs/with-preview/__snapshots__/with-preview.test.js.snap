// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withPreview works correctly withPreview is rendered correctly 1`] = `
<video
  height="175"
  muted={true}
  poster="img/the-grand-budapest-hotel-poster.jpg"
  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
  width="100%"
/>
`;
